src/lightning/fabric/utilities/spike.py

```diff
@@ -0,0 +1,149 @@
+import json
+import operator
+import os
+import warnings
+from typing import Any, Dict, List, Literal, Optional, Protocol, runtime_checkable, Union
+
+import torch
+from lightning_utilities.core.imports import compare_version
+
+_TORCHMETRICS_GREATER_EQUAL_1_0_0 = compare_version("torchmetrics", operator.ge, "1.0.0")
```

Comment:
Was there a problem with the regular RequirementCache? We should probably use it.